__tests__/app.test.js

```diff
@@ -125,6 +125,65 @@ describe("/articles/:article_id/comments", () => {
         });
       });
   });
+  test("POST:201 inserts a new comment of a given article_id to the db and sends the new comment back to the client for a username that is in the users table", () => {
+    const newComment = {
+      author: "butter_bridge",
+      body: "Lorem Ipsum is simply dummy text of the printing and typesetting industry.",
+    };
+    return req(app)
+      .post("/api/articles/6/comments")
+      .send(newComment)
+      .expect(201)
+      .then((res) => {
+        expect(res.body.comment).toHaveProperty("body");
+        expect(res.body.comment).toHaveProperty("votes");
+        expect(res.body.comment).toHaveProperty("author");
+        expect(res.body.comment).toHaveProperty("created_at");
+        expect(res.body.comment).toHaveProperty("comment_id");
+        expect(res.body.comment.article_id).toEqual(6);
+      });
+  });
+  test("POST:400 inserts a new comment of a given article_id to the db and sends the new comment back to the client for a username that is in the users table", () => {
+    const newComment = {
+      author: "tr-tobi",
+      body: "Lorem Ipsum is simply dummy text of the printing and typesetting industry.",
+    };
+    return req(app)
+      .post("/api/articles/6/comments")
+      .send(newComment)
+      .expect(400)
+      .then((res) => {
+        expect(res.body.msg).toBe("Invalid Username");
+      });
+  });
+
+  test("POST:404 sends an appropriate error message when given a valid but non-existent id", () => {
+    const newComment = {
+      author: "tr-tobi",
+      body: "Lorem Ipsum is simply dummy text of the printing and typesetting industry.",
+    };
+    return req(app)
+      .post("/api/articles/999/comments")
+      .send(newComment)
+      .expect(404)
+      .then((res) => {
+        expect(res.body.msg).toBe("Article does not exist");
+      });
+  });
+
+  test("POST:400 sends an appropriate and error message when given an invalid id", () => {
+    const newComment = {
+      author: "tr-tobi",
+      body: "Lorem Ipsum is simply dummy text of the printing and typesetting industry.",
+    };
+    return req(app)
+      .post("/api/articles/hello/comments")
+      .send(newComment)
+      .expect(400)
+      .then((res) => {
+        expect(res.body.msg).toBe("Invalid id");
+      });
```

Comment:
We should add a test for when we do not receive a username or a body, and send a 400 bad request 🙂